internal/introspection/router.go

```diff
@@ -24,26 +24,102 @@ import (
 	"go.uber.org/yarpc/api/transport"
 )
 
-// Procedure represent a registered procedure on a dispatcher.
-type Procedure struct {
-	Name      string `json:"name"`
-	Encoding  string `json:"encoding"`
-	Signature string `json:"signature"`
-	RPCType   string `json:"rpcType"`
-}
-
 // IntrospectProcedures is a convenience function that translate a slice of
 // transport.Procedure to a slice of introspection.Procedure. This output is
 // used in debug and yarpcmeta.
-func IntrospectProcedures(routerProcs []transport.Procedure) []Procedure {
+func IntrospectProcedures(routerProcs []transport.Procedure) Procedures {
 	procedures := make([]Procedure, 0, len(routerProcs))
 	for _, p := range routerProcs {
+		var spec interface{}
+		switch p.HandlerSpec.Type() {
+		case transport.Unary:
+			spec = p.HandlerSpec.Unary()
+		case transport.Oneway:
+			spec = p.HandlerSpec.Oneway()
+		}
+		var IDLEntryPoint *IDLModule
+		if spec != nil {
+			if i, ok := spec.(IntrospectableHandler); ok {
+				if i := i.Introspect(); i != nil {
+					IDLEntryPoint = i.IDLEntryPoint
+				}
+			}
+		}
 		procedures = append(procedures, Procedure{
-			Name:      p.Name,
-			Encoding:  string(p.Encoding),
-			Signature: p.Signature,
-			RPCType:   p.HandlerSpec.Type().String(),
+			Name:          p.Name,
+			Encoding:      string(p.Encoding),
```

Comment:
Similar note on typedefs as below (this isn't part of this PR but to point it out) - I don't like how we typedef'ed Encoding, instead of just using a string. It just leads to casting everywhere, and you don't get any real benefit, and if Encoding were to change later to an int, all the casts would be invalid.